### PR TITLE
vkd3d: Correctly handle output + fix NULL old targets for fullscreen 

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
The output here is actually for secure presentation and restricting a swapchain to a certain output.

Correctly handle NULL (desktop) targets that we used to have.

Fixes crashes with titles that use fullscreen via an initial fullscreen desc.

### CODE REVIEW

**Comment by Joshua-Ashton** [comment]
Actually, lemme do some extra cleanup.

File: libs/vkd3d/swapchain.c [hunk 1]
```diff
@@ -2421,23 +2416,25 @@ static HRESULT d3d12_swapchain_init(struct d3d12_swapchain *swapchain, IDXGIFact
         }
     }
 
-    IDXGIFactory_AddRef(swapchain->factory = factory);
-
-    if (FAILED(hr = d3d12_swapchain_set_fullscreen(swapchain, target, TRUE)))
+    if (FAILED(d3d12_swapchain_set_fullscreen(swapchain, target, TRUE)))
```
Comment (doitsujin) [inline libs/vkd3d/swapchain.c:2419]: `FAILED(hr = d3d12_swapchain_set_fullscreen`?

File: libs/vkd3d/swapchain.c [hunk 2]
```diff
@@ -2421,23 +2416,25 @@ static HRESULT d3d12_swapchain_init(struct d3d12_swapchain *swapchain, IDXGIFact
...
     {
-        WARN("Failed to go fullscreen.\n");
+        ERR("Failed to enter fullscreen.");
         return hr;
     }
 
-    if (!output && target)
+    if (!swapchain->fullscreen_desc.Windowed)
+        swapchain->target = target;
```
Comment (doitsujin) [inline libs/vkd3d/swapchain.c:2426]: Do we initialize swapchain->target anywhere if we don't take this code path?

Comment (Joshua-Ashton) [reply libs/vkd3d/swapchain.c:2426]: It is set to 0 already given it is calloc'ed

File: libs/vkd3d/swapchain.c (hunk 1 above)
Comment (Joshua-Ashton) [reply libs/vkd3d/swapchain.c:2419]: It should, good catch.

**Pull request merged**